tests/rbd_mirror/test_rbd_mirror_connection_metrics.py

```diff
@@ -0,0 +1,149 @@
+"""Test case covered -
+    CEPH-83575566 - Performance counter for rbd mirror network
+    connection health metrics.
+
+    Pre-requisites :
+    1. Two Clusters must be up and running to create pool
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files on each node.
+
+    Test Case Flow:
+        1. Create a pool on both clusters.
+        2. Create an Image on primary mirror cluster in same pool.
+        3. Configure mirroring (peer bootstrap) between two clusters.
+        4. Enable image mode snapshot based mirroring on the pool respectively.
+        5. Start running IOs on the primary image.
+        6. Verify ceph exporter from prometheus for network connection
+           health metrics.
+"""
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def validate_connection_metrics(admin_node, port):
+    """Method to validate network connection health metrics.
+    Args:
+        admin_node: cephadm node with ceph-exporter deployed.
+        port: ceph-exporter port number to get metrics
+
+    Example:
+        # TYPE ceph_AsyncMessenger_Worker_msgr_connection_idle_timeouts counter
+        ceph_AsyncMessenger_Worker_msgr_connection_idle_timeouts{id="0"} 128
+
+        # TYPE ceph_AsyncMessenger_Worker_msgr_connection_ready_timeouts counter
+        ceph_AsyncMessenger_Worker_msgr_connection_ready_timeouts{id="0"} 0
+
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+    # connection metrics to validate
+    connection_metrics = [
+        "ceph_AsyncMessenger_Worker_msgr_connection_idle_timeouts",
+        "ceph_AsyncMessenger_Worker_msgr_connection_ready_timeouts",
+    ]
+
+    node_ip = admin_node.ip_address
+    cmd = f"curl http://{node_ip}:{port}/metrics | grep ceph_AsyncMessenger_Worker_msgr"
+
+    out = admin_node.exec_command(cmd=cmd, output=True, sudo=True)
+    lines = out[0].split("\n")
+
+    # Dictionary to store parsed metrics
+    metrics = {}
+
+    # Iterate over lines and extract metrics
+    for line in lines:
+        if line.startswith("#") or not line.strip():
+            # Skip comments and empty lines
+            continue
+
+        # Split the line into parts using whitespace
+        parts = line.split()
+
+        # Extract relevant information
+        metric_name = parts[0]
+        metric_value = parts[1]
```

Comment:
You could actually reduce these 3 lines into a single line like this
[metric_name, metric_value] = line.split()

Reply:
Done